capa/features/extractors/ghidra/file.py

```diff
@@ -0,0 +1,198 @@
+    """
+    for seg in currentProgram.getMemory().getBlocks():
+        for ea, _ in check_segment_for_pe(seg):
+            yield Characteristic("embedded pe"), FileOffsetAddress(ea)
+
+
+def extract_file_export_names() -> Iterator[Tuple[Feature, Address]]:
+    """extract function exports
+
+    Exports in Ghidra:
+        - namespace = Global
+        - SourceType = IMPORTED
+    """
+
+    fm = currentProgram.getFunctionManager().getFunctions(True) # Only Global functions
+    while fm.hasNext():
+        f = fm.next().getSymbol()
+        if (f.getSource() == SourceType.IMPORTED):
+            yield Export(f.toString()), AbsoluteVirtualAddress(f.getAddress().getOffset())
+
+
+def extract_file_import_names() -> Iterator[Tuple[Feature, Address]]:
+    """extract function imports
+
+    1. imports by ordinal:
+     - modulename.#ordinal
+
+    2. imports by name, results in two features to support importname-only
+       matching:
+     - modulename.importname
+     - importname
+    """
+
+    fm = currentProgram.getFunctionManager().getExternalFunctions()
+    while fm.hasNext():
+        f = fm.next()
+        addr = int(f.getEntryPoint().toString().split(':')[1], 16)  # format: EXTERNAL:<hex_addr>
+        fstr = f.getName()  # format: 'importname' / 'Ordinal_*'
+        if 'Ordinal_' in fstr:
+            fstr = f"#{fstr.split('_')[1]}"
+        yield Import(fstr), AbsoluteVirtualAddress(addr)
+
+
+def extract_file_section_names() -> Iterator[Tuple[Feature, Address]]:
+    """extract section names"""
+
+    for block in currentProgram.getMemory().getBlocks():
+        yield Section(block.getName()), AbsoluteVirtualAddress(block.getStart().getOffset())
+
+
+def extract_file_strings() -> Iterator[Tuple[Feature, Address]]:
+    """extract ASCII and UTF-16 LE strings"""
+
+    dat = getFirstData()
+    while (dat != None):
+        if (dat.hasStringValue()):
+            yield String(dat.getValue()), FileOffsetAddress(dat.getAddress().getOffset()) 
+        dat = getDataAfter(dat)
+
+
+def extract_file_function_names() -> Iterator[Tuple[Feature, Address]]:
+    """
+    extract the names of statically-linked library functions.
+    """
+
+    fm = currentProgram.getFunctionManager().getExternalFunctions()
+    while fm.hasNext():
+        f = fm.next()
+        addr = int(f.getEntryPoint().toString().split(':')[1], 16)  # format: EXTERNAL:<hex_addr>
+        name = f.getName()
+        yield FunctionName(name), AbsoluteVirtualAddress(addr)
+        if name.startswith("_"):
+            # some linkers may prefix linked routines with a `_` to avoid name collisions.
+            # extract features for both the mangled and un-mangled representations.
+            # e.g. `_fwrite` -> `fwrite`
+            # see: https://stackoverflow.com/a/2628384/87207
+            yield FunctionName(name[1:]), AbsoluteVirtualAddress(addr)
+
+
+def extract_file_format() -> Iterator[Tuple[Feature, Address]]:
+
+    ef = currentProgram.getExecutableFormat()
```

Comment:
Do we have a good idea of the variations returned by `getExecutableFormat`?

Reply:
Here are the returned strings via PE, ELF, and Raw samples from our test files:
```
'Executable and Linking Format (ELF)'
'Portable Executable (PE)'
'Raw Binary'
```